frontend/lib/src/components/shared/Tooltip/Tooltip.test.tsx

```diff
@@ -15,54 +15,63 @@
  */
 
 import React from "react"
-import { PLACEMENT } from "baseui/tooltip"
-import { mount } from "@streamlit/lib/src/test_util"
+import "@testing-library/jest-dom"
+import { screen, fireEvent } from "@testing-library/react"
+import { BaseProvider, LightTheme } from "baseui"
+import { render } from "@streamlit/lib/src/test_util"
 
 import Tooltip, { Placement, TooltipProps } from "./Tooltip"
 
 const getProps = (
   propOverrides: Partial<TooltipProps> = {}
 ): TooltipProps => ({
   placement: Placement.AUTO,
-  content: () => {},
+  content: <div>Tooltip content text.</div>,
   children: null,
   ...propOverrides,
 })
 
+// Wrap in BaseProvider to avoid warnings
+const renderTooltip = (props: Partial<TooltipProps> = {}): any => {
+  return render(
+    <BaseProvider theme={LightTheme}>
+      <Tooltip {...getProps(props)} />
+    </BaseProvider>
+  )
+}
+
 describe("Tooltip element", () => {
-  it("renders a Tooltip", () => {
-    const wrapper = mount(<Tooltip {...getProps()}></Tooltip>)
+  it("renders a Tooltip", async () => {
+    renderTooltip()
+
+    const tooltipTarget = screen.getByTestId("tooltipHoverTarget")
+    expect(tooltipTarget).toBeInTheDocument()
 
-    expect(wrapper.find("StatefulTooltip").exists()).toBeTruthy()
+    // Hover to see tooltip content
+    fireEvent.mouseOver(tooltipTarget)
+
+    const tooltipContent = await screen.findByTestId("stTooltipContent")
+    expect(tooltipContent).toHaveTextContent("Tooltip content text.")
   })
 
   it("renders its children", () => {
-    const wrapper = mount(
-      <Tooltip {...getProps()}>
-        <div className="foo" />
-      </Tooltip>
-    )
+    renderTooltip({ children: <div>Child Element</div> })
 
-    expect(wrapper.find(".foo").exists()).toBeTruthy()
+    expect(screen.getByTestId("tooltipHoverTarget")).toBeInTheDocument()
+    expect(screen.getByText("Child Element")).toBeInTheDocument()
   })
 
-  it("sets its placement", () => {
```

Comment:
Removed this test - since couldn't find checkable style resulting from placement prop & placement covered by E2E